File: algorithms/209. Minimum Size Subarray Sum/solution.py
class Solution(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        left = total = 0
        minlen = float('inf')
        for right in xrange(len(nums)):
            total += nums[right]
            while total >= s + nums[left]:
                total -= nums[left]
                left += 1
            if total >= s:
                minlen = min(minlen, right - left + 1)
        return minlen if minlen != float('inf') else 0

File: algorithms/209. Minimum Size Subarray Sum/solution.cc
class Solution {
public:
    int minSubArrayLen(int s, vector<int>& nums) {
        int m = INT_MAX, n = nums.size(), sum = 0;
        for ( int i = 0, j = 0; j < n; ++j ) {
            sum += nums[j];
            while ( sum >= s + nums[i] )
                sum -= nums[i++];
            if ( sum >= s )
                m = min(m, j - i + 1);
        }
        return m == INT_MAX ? 0 : m;
    }
};
